framework/source/class/qx/ui/window/Window.js

```diff
@@ -632,17 +679,34 @@ qx.Class.define("qx.ui.window.Window",
         this.hide();
         this.fireEvent("close");
       }
+
+      // If automatically destroying the window upon close was requested, do
+      // so now. (Note that we explicitly re-obtain the autoDestroy property
+      // value, allowing the user's close handler to enable/disable it before
+      // here.)
+      if (this.getAutoDestroy()) {
+        this.removeAll();                                    // remove contents
+        qx.event.Registration.removeAllListeners(this);      // remove listeners
+        qx.core.Init.getApplication().getRoot().remove(this);// remove self
+        this.dispose();
+      }
     },
 
 
     /**
-     * Opens the window.
+     * Open the window. If the {@link qx.ui.win.Window#autoCenter} property is
+     * true, also center the window.
      */
     open : function()
     {
       this.show();
       this.setActive(true);
       this.focus();
+
+      // If autoCenter is true, then center the window
+      if (this.getAutoCenter()) {
+        this.center();
```

Comment:
The same with our code. I've created a mixin which adds a ```centered``` property and adds an event listener to both ```resize``` and ```appear``` event.
Additionally I've added an applier to the ```centered``` property which centers the window when set to true while the window is open.